Create database schema, populate it and add basic endpoints

vdrelciuc opened the pull request:
This PR contains:
- SQL schema and mock data (a few basic rooms with questions, answers and participants)
- Common interfaces that can be used both by the client and the server (they define the data structure)
- Database debugging endpoints to list all elements of a table and to reset the mock data
- 3 basic endpoints: get all questions given an event key, get all choices given a question id, post an answer given question id and participant id
- Updated README.MD to include instructions on how to set up the database locally for testing purposes
Warning: I'll need to provide you a .env file containing all the API keys before you'll be able to access the database

nitinms-sps-program commented on server/app/services/database.service.ts:
```diff
@@ -0,0 +1,62 @@
+    async createAnswer(questionId: string, participantId: string, answerLabel: string) {
+        return this.pool.query(`
+        INSERT INTO ${this.SCHEMA_NAME}.AnswerEntry (question_id, participant_id, answer_label) 
+        VALUES (${questionId},${participantId},'${answerLabel}');`);
```
What happens if answerLabel has a quote in it?

vdrelciuc replied:
I'm working on a way to sanitize user input on the server side. I'll update my PR as soon as I get it working

vdrelciuc replied:
Fixed in commit 6584c0eed

vdrelciuc commented on README.md:
```diff
@@ -1,11 +1,24 @@
+2. After selecting the right project, navigate to SQL
+3. Select the right instance
+4. Under Connections -> Authorized networks, add your IP Address
+5. Don't forget to upload your .env file to /server
```
Copying the .env file inside the local checkout, yes :)

vdrelciuc commented on server/app/controllers/choice.controller.ts:
```diff
@@ -0,0 +1,34 @@
+                    res.status(HttpStatus.OK).send(choices);
+                })
+                .catch((e: Error) => {
+                    res.status(HttpStatus.BAD_REQUEST).send(e.message);
```
It won't, we'll deal with the front-end error management using the HTTP status codes (e.g. telling the user that one of his entries is incorrect). The e.message is only for debugging purposes. Would you like me to take it off completely?

amulyaparmar commented on common/interfaces/quiz.ts:
```diff
@@ -0,0 +1,5 @@
+export interface Quiz {
+    quizId: number;
+    maxDuration: number;
+    roomId: number;
```
What happens if the quiz does not have a Room ID?

vdrelciuc replied:
Depends on the endpoint implementation. I haven't had the chance to code an endpoint that needed a quiz ID so far, but if I were to write one, it'd send a 400 BAD REQUEST answer to the client if it was missing arguments

vdrelciuc commented:
I've added unit tests for both the controllers and the database service. I also fixed the SQL injection vulnerability by using parameterized queries. This also takes care of input sanitization (such as single quotes and double quotes).
